File: task_priority_manager/python/task_priority_manager.py
from datetime import datetime
from typing import List, Optional, Dict

class Task:
    def __init__(self, id: int, title: str, description: str, priority: int):
        self.id = id
        self.title = title
        self.description = description
        self.priority = priority
        self.created_at = datetime.now()
        self.completed = False

class TaskPriorityManager:
    def __init__(self):
        self.tasks: Dict[int, Task] = {}
        self.next_id = 1

    def add_task(self, title: str, description: str, priority: int) -> Task:
        # TODO: Implement this method
        pass

    def update_task_priority(self, task_id: int, new_priority: int) -> bool:
        # TODO: Implement this method
        pass

    def delete_task(self, task_id: int) -> bool:
        # TODO: Implement this method
        pass

    def get_next_task(self) -> Optional[Task]:
        # TODO: Implement this method
        pass

    def list_tasks(self) -> List[Task]:
        # TODO: Implement this method
        pass

def run_tests():
    task_manager = TaskPriorityManager()
    
    # Test 1: Add tasks
    task_manager.add_task("Fix critical bug", "Fix the login issue", 1)
    task_manager.add_task("Update documentation", "Update API docs", 3)
    task_manager.add_task("Code review", "Review PR #123", 2)
    
    # Test 2: Get next task
    next_task = task_manager.get_next_task()
    assert next_task.priority == 1, "Next task should be highest priority"
    
    # Test 3: Update priority
    task_manager.update_task_priority(2, 1)
    updated_task = task_manager.tasks[2]
    assert updated_task.priority == 1, "Task priority should be updated"
    
    # Test 4: List tasks
    all_tasks = task_manager.list_tasks()
    assert len(all_tasks) == 3, "Should return all tasks"
    assert all_tasks[0].priority <= all_tasks[1].priority, "Tasks should be sorted by priority"
    
    # Test 5: Delete task
    task_manager.delete_task(1)
    assert 1 not in task_manager.tasks, "Task should be deleted"
    
    print("All tests passed!")

if __name__ == "__main__":
    # Uncomment to run tests
    # run_tests()
    pass 

File: task_priority_manager/javascript/task_priority_manager.js
class Task {
  constructor(id, title, description, priority) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.priority = priority;
    this.created_at = new Date();
    this.completed = false;
  }
}

class TaskPriorityManager {
  constructor() {
    this.tasks = new Map();
    this.nextId = 1;
  }

  addTask(title, description, priority) {
    // TODO: Implement this method
  }

  updateTaskPriority(taskId, newPriority) {
    // TODO: Implement this method
  }

  deleteTask(taskId) {
    // TODO: Implement this method
  }

  getNextTask() {
    // TODO: Implement this method
  }

  listTasks() {
    // TODO: Implement this method
  }
}

// Test cases
function runTests() {
  const taskManager = new TaskPriorityManager();

  // Test 1: Add tasks
  taskManager.addTask("Fix critical bug", "Fix the login issue", 1);
  taskManager.addTask("Update documentation", "Update API docs", 3);
  taskManager.addTask("Code review", "Review PR #123", 2);

  // Test 2: Get next task
  const nextTask = taskManager.getNextTask();
  console.assert(
    nextTask.priority === 1,
    "Next task should be highest priority"
  );

  // Test 3: Update priority
  taskManager.updateTaskPriority(2, 1);
  const updatedTask = taskManager.tasks.get(2);
  console.assert(updatedTask.priority === 1, "Task priority should be updated");

  // Test 4: List tasks
  const allTasks = taskManager.listTasks();
  console.assert(allTasks.length === 3, "Should return all tasks");
  console.assert(
    allTasks[0].priority <= allTasks[1].priority,
    "Tasks should be sorted by priority"
  );

  // Test 5: Delete task
  taskManager.deleteTask(1);
  console.assert(!taskManager.tasks.has(1), "Task should be deleted");

  console.log("All tests passed!");
}

// Uncomment to run tests
// runTests();
